tests/ArmTemplates.Tests/Extractor/Scenarios/ProductApisExtractorTests.cs

```diff
@@ -64,7 +64,7 @@ public async Task GenerateProductApisTemplates_ProperlyLaysTheInformation()
                 this.GetTestLogger<ExtractorExecutor>(),
                 null, null, null, null, null, null, null,
                 productApisExtractor: productApisExtractor,
-                null, null, null, null);
+                null, null, null, null, null);
```

Comment:
We might want to centralize this for all tests to reduce duplication.

Idea would be to have a `BuildExtractorExecutor` with all properties as optional parameters (default `null`) and then people can provide the one they need to build an instance.

That way, all new additions are just in that one method instead of across methods

Reply:
done